signer/src/storage/model.rs

```diff
@@ -163,6 +163,20 @@ pub struct Transaction {
     pub created_at: time::OffsetDateTime,
 }
 
+/// Persisted DKG shares
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "testing", derive(fake::Dummy))]
+pub struct EncryptedDkgShares {
```

Comment:
>  since it is only one of many private DKG shares.

Actually, maybe not. I suppose this can contain the aggregated secret shares (where each received secret share has been multiplied by the appropriate coefficient). Then you just store one aggregate secret key share and represent it by this struct.

Reply:
Just rechecked this, and this does indeed store all the DKG private shares as the `SignerState`, just encrypted. Can we update the comment of the `encrypted_shares` field to note this fact.

Reply:
I'm not sure I follow which coefficient you refer to here, each signer in v1 will hold a single secret share but in general WSTS allows a single party to hold multiple shares of the aggregate key corresponding to the weight of that party. In either case, a single share is a sum of the evaluations of all private polynomials (each evaluation is called a private share in wsts, so there's a bit of confusing reuse of terminology here) at a specific point. And afaik we store the sum, not the evaluations if that's what you're getting at.

But since a single party may hold multiple shares, I think it makes sense to use the plural form and call it `EncryptedDkgShares`.

Reply:
> I'm not sure I follow which coefficient you refer to here, each signer in v1 will hold a single secret share but in general WSTS allows a single party to hold multiple shares of the aggregate key corresponding to the weight of that party. In either case, a single share is a sum of the evaluations of all private polynomials (each evaluation is called a private share in wsts, so there's a bit of confusing reuse of terminology here) at a specific point. And afaik we store the sum, not the evaluations if that's what you're getting at.

Oh, I was going a step further. I think we're starting on the same page, where each secret share $`s_i`$ takes the form
```math
s_i = \sum_{k=1}^n f_k(i)
```
and that we store the sum, and that each party can have multiple secret shares. But I was thinking about what we need for signing. Specifically, the $`i`$-th signing share takes the form:
```math
z_{i} = d_{i} + \rho_{i} \cdot e_{i} + c \cdot \sum_{k \in K_{i}}^{} \lambda_{k} \cdot s_{k}.
```
where $`d_i`$ and $`e_i`$ are private nonces and $`\rho_i`$ and $`c`$ are hashes of stuff. I was thinking that the $`i`$-th signer was just storing $`\sum \lambda_{k} \cdot s_{k}`$, since that can be precomputed.

> But since a single party may hold multiple shares, I think it makes sense to use the plural form and call it `EncryptedDkgShares`.

Yeah that makes sense, let's keep it as is. I was going with the unintuitive interpretation that $`\sum \lambda_{k} \cdot s_{k}`$ is the secret share that happens to have more weight that other secret shares. But that interpretation would be an abuse of terminology.

---

But I was just hypothesizing what was going on here, not suggesting that we do that. I think that it's far simpler and better to go with what you have here.

Reply:
Alright, thanks for elaborating. And nice observation. I think it would be a good optimization to add down the line. I suppose this could be updated in `wsts` without any changes to the signer code. Although if we change the representation of the stored data we'd need to think about how to handle migrations for this.

Reply:
Yeah, that will be an issue. In general, I kinda think we should store `protobufs` for the ease of upgradability. But I'm not entirely convinced I can make the case.

Reply:
Yeah, I'm not sure I can motivate that extra effort right now either.